### PR TITLE
Draft - Vault Connector resources

### PR DESCRIPTION
**Pull request by zhcli**
This is a draft PR in regards to - https://github.com/harness/terraform-provider-harness/issues/146. Both resources were tested locally against a test harness instance. 

It is depending on how harness-go-sdk is going to expose Vault type in ConnectorInfo type, here is my tested changes - https://github.com/harness/harness-go-sdk/compare/main...zhcli:harness-go-sdk:main.

Documents and related Data resources will be added once the community has agreed on this approach.

Added:
- Resource for Vault Connector with Vault agent authentication method
- Resource for Vault Connector with K8s authentication method

Updated:
- connector.go: fixing connector not found error when deleting connectors that are created at Organisation or Project level

Hi @micahlmartin @rijajoo , could you please review and provide direction?
FYI @hwh57

### CODE REVIEW

**Review by FireGrey — COMMENTED** [review]
Just noticed you included a fix for connector deletion before I made a seperate PR for it in #190

**Comment by zhcli** [comment]
> 

I will merge your changes once your PR is in. My PR is still in draft, awaiting harness direction.

**Comment by micahlmartin** [comment]
@rathodmeetsatish I think we may need to regen the `harness-go-sdk` to make sure we get the vault information in here. Can you look into this when you get a chance?

File: internal/service/platform/connector/connector.go [hunk 1]
```diff
@@ -83,7 +83,19 @@ func resourceConnectorCreateOrUpdateBase(ctx context.Context, d *schema.Resource
...
+	if ok && orgIdentifier != nil {
+		localVarOptionals.OrgIdentifier = optional.NewString(orgIdentifier.(string))
+	}
+
+	projIdentifier, ok := d.GetOk("project_id")
+	if ok && orgIdentifier != nil {
+		localVarOptionals.ProjectIdentifier = optional.NewString(projIdentifier.(string))
+	}
+
+	_, _, err := c.ConnectorsApi.DeleteConnector(ctx, c.AccountId, d.Id(), localVarOptionals)
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/connector.go:98]: We dont need to do this as it is fixed in an earlier pr

File: internal/service/platform/connector/hcv_vault_agent.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+func buildConnectorHCVAgent(d *schema.ResourceData) *nextgen.ConnectorInfo {
+	connector := &nextgen.ConnectorInfo{
+		Type_: nextgen.ConnectorTypes.Vault,
+		Vault: &nextgen.VaultConnector{},
+	}
+
+	connector.Vault.UseVaultAgent = true
+	connector.Vault.SecretEngineVersion = 2
+	connector.Vault.AccessType = "VAULT_AGENT"
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/hcv_vault_agent.go:111]: This pr looks only for vault agent authentication mechanism the pattern we are following is for a particular secret manager we have only one resource

Comment (rathodmeetsatish) [reply internal/service/platform/connector/hcv_vault_agent.go:111]: So for this to be merged we'll have to modify the resource to support all authentication mechanism.You can refer sshkey secret resource https://github.com/harness/terraform-provider-harness/blob/main/internal/service/platform/secret/sshkey.go

Comment (zhcli) [reply internal/service/platform/connector/hcv_vault_agent.go:111]: Thanks, @rathodmeetsatish for your feedback. I will merge both resources into one in alignment with Harness's direction when I have some spare time during the weekend.

**Comment by rathodmeetsatish** [comment]
closing it as vault connector is added in this pr https://github.com/harness/terraform-provider-harness/pull/250

**Pull request closed without merging**